lib/BuildSystem/LaneBasedExecutionQueue.cpp

```diff
@@ -106,6 +113,22 @@ class LaneBasedExecutionQueue : public BuildExecutionQueue {
     }
   }
 
+  void killAfterTimeout() {
+      ::sleep(10);
+
+      if (!done) {
```

Comment:
True, the `done` flag was actually supposed to counter the use-after-free, but now I realise that doesn't actually make sense and just worked by accident :)